[READY] Adds Uplifted Monkeys

Zero-Bits opened the pull request:
:cl:ZeroBits
rscadd: "Added new species: Uplifted Primates."
tweak: "Voicebox is now a general organ which other species' inherit from."
bugfix: "Body-hair code now works properly."
/:cl:

What's left?
- [x] Spritework
- [x] Lore
- [x] Steal Resomi code for fur color
- [x] Define job restrictions
- [x] Appease Travis

Requested Changes:
- [X] Make voicebox a general organ and inherit from it
- [X] Change name to a define

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Zero-Bits committed:
Adds Uplifted Monkeys

ZCaliber commented:
Sounds pretty cool. Might cause some hubba baloo with people using lab chimps as food for slimes and experimentation.

Also: https://youtu.be/tLFqxjpACeg

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/species/station/chinsky.dm:
```diff
@@ -0,0 +1,38 @@
+/datum/species/human/uplift
+	name = "Uplift"
```
Needs to be a define

PurpleMartinJCK commented on code/modules/organs/subtypes/chinsky.dm:
```diff
@@ -0,0 +1,15 @@
+
+/obj/item/organ/internal/voicebox/monkey
```
This is copy paste from here. https://github.com/Baystation12/Baystation12/blob/dc176c5742694b2b641d53dfa08a3daf0ccce2de/code/modules/organs/subtypes/nabber.dm#L2 and this instead should be made into a new general organ they both can inherit from.

Zero-Bits replied:
I'll see about doing that

atlantiscze commented:
Given this is literally a new species, i would like to request a lore approval, as well as some additional information about what the lore and possible special mechanics for this species are going to be.

Zero-Bits commented:
@atlantiscze I'm working with lone directly on lore. It's not yet in a presentable state.

PurpleMartinJCK commented:
You thought about using the old resomi clothing sprites?

Zero-Bits commented:
I believe that any method requiring another set of clothing sprites is just going to be needless bloat. I'll likely just make them compatible with human clothing.

PurpleMartinJCK commented:
So they're going to be shaped like the current humans? I don't see how that could be done with still looking decent. Also are they going to be small mobs? If so it won't work well with existing sprites.

Nerev4r commented:
I don't necessarily agree with them being man shaped.

PsiOmegaDelta commented on code/modules/organs/subtypes/chinsky.dm:
```diff
@@ -0,0 +1,15 @@
+	will_assist_languages = list(LANGUAGE_GALCOM, LANGUAGE_LUNAR, LANGUAGE_SOL_COMMON, LANGUAGE_INDEPENDENT, LANGUAGE_SPACER)
+
+
+/obj/item/organ/internal/voicebox/monkey/New()
```
Non-datum `/New()` overrides shall always call ..()

Zero-Bits commented:
Either person-shaped or with code modifications for clothing offsets.

Zero-Bits commented:
Possibly something like this. ![preview](https://user-images.githubusercontent.com/7545411/32976030-605684b8-cbd4-11e7-9f7d-774abeaf9d4b.png)

Zero-Bits committed:
commit

Zero-Bits commented:
It is not ready for review.

Zero-Bits committed:
*evil cackling*

Zero-Bits committed:
Fix bodyhair code

Zero-Bits commented:
The CODE is ready for review. Still, have to make the sprites and handle lore. It has been tested, and it all works beautifully code-wise.

Zero-Bits commented:
Squash will happen when the sprites are committed.

Zero-Bits committed:
mlem

BlueNexus commented:
I don't like the current sprites. Too human-shaped.

Zero-Bits commented:
@BlueNexus And I don't like making sprites. If you can find someone who wants to make all the species' and clothing sprites relatively quickly, please, for the love of god, send them my way. Because that would be far better than me spriting it. But until that happens, we're stuck with whatever I can/want to make.

Zero-Bits commented:
In fact, here's a list of what we'd need before the species is actually usable conveniently. I'll put the species on hold for you until they're all finished. So you don't have to suffer through my subpar sprites.

General:
- [ ] Generic Undersuit
- [ ] Utility (Contractor) Jumpsuit
- [ ] Hazard Vests
- [ ] Emergency Softstuit
- [ ] Emergency Helmet
- [ ] Breath Mask
- [ ] Half-Mask
- [ ] Gas-Mask
- [ ] Caps
- [ ] Other Misc Hats
- [ ] Shoes or Sandals
- [ ] Work/Jackboots
- [ ] Backpack
- [ ] Satchel
- [ ] Messenger Bag
- [ ] Dufflebag
- [ ] Leather Satchel
- [ ] Black Satchel
- [ ] Leather Jacket
- [ ] Black Leather Jacket
- [ ] Bomber Jacket
- [ ] Work Gloves
- [ ] Colored Gloves
- [ ] Evening Gloves

Engineering Contractor/Roboticist:
- [ ] Hardhats
- [ ] Robotics Jumpsuit
- [ ] Hazard Jumpsuit
- [ ] Toolbelt
- [ ] Engineer Hardsuit
- [ ] Heavy Engineer Hardsuit
- [ ] Atmos Hardsuit
- [ ] Heavy Atmos Hardsuit
- [ ] Radiation Suit
- [ ] Meson Scanners
- [ ] Welding Helmet
- [ ] Welding Goggles
- [ ] Superior Welding Goggles
- [ ] Chief Engineer Hardsuit
- [ ] Engineering Backpack
- [ ] Engineering Satchel
- [ ] Engineering Messenger Bag
- [ ] Engineering Dufflebag
- [ ] Insulated Gloves

Cargo Contractor:
- [ ] Cargo Jumpsuit
- [ ] Cargo Cap

Janitor:
- [ ] Janitor's Jumpsuit
- [ ] Galoshes
- [ ] Trashbag
- [ ] Biosuit
- [ ] Biosuit Hood

Miner:
- [ ] Mining Jumpsuit
- [ ] Salvage Hardsuit
- [ ] Mining Hardsuit

Lab Assistant/NT Pilot:
- [ ] NT Undersuit
- [ ] Labcoat
- [ ] NT Labcoat
- [ ] Colored Labcoat
- [ ] Xenobio Biosuit
- [ ] Xenobio Biosuit Hood
- [ ] Anomaly Suit
- [ ] NanoTrasen Messenger Bag
- [ ] NanoTrasen Satchel
- [ ] NanoTrasen Backpack
- [ ] NanoTrasen Pilot Jumpsuit (also the leather jacket mentioned earlier)

Medical Contractor:
- [ ] Orderly's Jumpsuit
- [ ] Blue Scrubs
- [ ] Green Scrubs
- [ ] Black Scrubs
- [ ] Purple Scrubs
- [ ] Caps for the last 4
- [ ] Surgical Apron
- [ ] Latex/Nitrile Gloves

Other:
- [ ] Chef Undersuit
- [ ] Chef's Apron
- [ ] Chef's Hat
- [ ] Bartender Undersuit
- [ ] Hydroponicist's Suit
- [ ] Overalls

hey-it-stev commented:
Welp, someone's gonna have to make an executive decision about this at some point. Implement with human-sized sprites and have it ready soon-ish or get a ton of people to do new clothing sprites and maybe have it at some point.

Zero-Bits commented:
Yup. Until that decision is made, this is on hold.

Zero-Bits committed:
Uplifts are now cute, lovable, fun monkeys

This is what you asked for. Don't blame me when it's Resomi 2; just
reskinned as monkeys.

Zero-Bits commented:
Alright, fine. Here you go. I'm done with this. You wanted this, so here it is. No, I will not fix [x] sprite issue. No, I will not make new clothing. It was good enough for monkeys, so it's good enough for uplifts. No, I'm not to blame when they're literally Resomi 2. Yes, my jimmies are rustled. Yes, I think this is stupid.
![ook](https://user-images.githubusercontent.com/7545411/33338041-736c30ac-d43a-11e7-9050-dac61859556e.png)

comma commented:
Calm down, jesus.

hey-it-stev commented:
Now that we're no longer using human-size sprites we're going to need voidsuit sprites and probs a new refitted one to switch to.

Zero-Bits committed:
Travis

Zero-Bits commented:
Not my problem.

Zero-Bits commented:
Hell, none of this is my problem. I was doing this for fun, and because I wanted to. It's no longer fun, and I no longer want to.
